vmware/ngc/NGC-AdapterManager/src/main/java/org/opensds/vmware/ngc/adapters/opensds/RestClient.java

```diff
@@ -291,31 +327,96 @@ JSONObject getStoragePool(String poolId) throws Exception {
         return response;
     }
 
-    JSONObject attachVolume(String volumeId, String initiator, String initiatorIp) throws Exception {
-        JSONObject hostInfo = new JSONObject();
-        hostInfo.put("initiator", initiator);
-        hostInfo.put("ip", initiatorIp);
+    JSONObject getHost(String hostName) throws Exception {
+        logger.info(String.format("----------OpenSDS Getting Host %s----------", hostName));
+
+        JSONArray response = (JSONArray)request.get(String.format("/host/hosts?hostName=%s", hostName));
+        logger.debug(String.format("OpenSDS Getting Host %s Response: %s", hostName,
+                    response));
 
+        if (response.isEmpty()) {
+            String msg = String.format("No Host Found");
+            logger.info(String.format("OpenSDS Get Host for %s Error: %s", hostName, msg));
+            return new JSONObject();
+        }
+
+        JSONObject host = (JSONObject) response.get(0);
+        return host;
+    }
+
+    JSONObject createHost(ConnectMO connect) throws Exception {
         JSONObject requestData = new JSONObject();
-        requestData.put("volumeId", volumeId);
-        requestData.put("hostInfo", hostInfo);
+        requestData.put("hostName", connect.name);
+        requestData.put("ip", connect.initiatorIp);
+        requestData.put("osType", connect.osType.toString().toLowerCase());
+        requestData.put("accessMode", OPENSDS_HOST_ACCESSMODE.getValue());
+        String availabilityZone = OPENSDS_AVAILABILITYZONE.getValue();
+
+        if(!availabilityZone.isEmpty()) {
+			JSONArray availabilityZones = new JSONArray();
+			availabilityZones.put(availabilityZone);
+			requestData.put("availabilityZones", availabilityZones);
+		}
```

Comment:
Any indentation issue here?

Reply:
What happens if availabilityZone is empty?

Reply:
Ok, maybe this will throw error in POST and error handled at 374 line

Reply:
Yes, handled